walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,43 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.*;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> results = new ArrayList<TimeRange>();
+
+    // Find conflicts.
+    ArrayList<TimeRange> conflicts = new ArrayList<TimeRange>();
+    for (Event event : events) {
+      if (!Collections.disjoint(event.getAttendees(), request.getAttendees())) {
+        conflicts.add(event.getWhen());
+      }
+    }
+    Collections.sort(conflicts, TimeRange.ORDER_BY_START);
+    Iterator<TimeRange> conflictIterator = conflicts.iterator();
+
+    // Find possible meeting times.
+    long duration = request.getDuration();
+    long currentTime = Long.valueOf(TimeRange.START_OF_DAY);
+    TimeRange previous = TimeRange.fromStartDuration(TimeRange.START_OF_DAY, 0);
+    while (conflictIterator.hasNext()) {
+      TimeRange conflictTime = conflictIterator.next();
+      if (previous.contains(conflictTime)) {
+        continue;
+      }
+      long start = Long.valueOf(conflictTime.start());
+      if ((start > currentTime) && ((start - currentTime) >= duration)) {
```

Comment:
```suggestion
      if ((start - currentTime) >= duration)) {
```

This should be enough to cover it right? Assuming `duration` isn't ever negative.

Reply:
For sure!